app/controllers/agents_controller.rb

```diff
@@ -237,14 +240,14 @@ def redirect_back(message, options = {})
       if @agent && !@agent.destroyed?
         path = agent_path(@agent)
       end
-    when /\A#{Regexp::escape scenarios_path}\/\d+\Z/, agents_path
+    when /\A#{Regexp::escape scenarios_path}\/\d+\z/, agents_path
       path = ret
     end
 
     if path
       redirect_to path, notice: message
     else
-      super agents_path, notice: message
```

Comment:
I think this `super` was necessary to get you back to the page you came from when you hit a button on the second page of your agents list, for example. (Sorry if I didn't add the spec for that)